src/vcpkg/postbuildlint.cpp

```diff
@@ -520,17 +519,17 @@ namespace vcpkg::PostBuildLint
     }
 
     static LintStatus check_dll_architecture(const std::string& expected_architecture,
-                                             const std::vector<path>& files,
+                                             const std::vector<Path>& files,
                                              const Filesystem& fs)
     {
         std::vector<FileAndArch> binaries_with_invalid_architecture;
 
-        for (const path& file : files)
+        for (const Path& file : files)
         {
             Checks::check_exit(VCPKG_LINE_INFO,
-                               Strings::case_insensitive_ascii_equals(vcpkg::u8string(file.extension()), ".dll"),
+                               Strings::case_insensitive_ascii_equals(file.extension(), ".dll"),
```

Comment:
Extension comparison

Reply:
I'm not sure what you want here. We have NotExtensionCaseSensitive and NotExtensionCaseInsensitive for use as predicates but I don't have a clearer way to say that here.